File: lab_2/generate/generator.cpp
#include <iostream>
#include <random>
#include <string>

int main() {
  std::random_device rd;
  std::mt19937 gen(rd());
  std::uniform_int_distribution<> dis(0, 1);

  std::string sequence = "";
  for (int i = 0; i < 128; ++i) {
    sequence += std::to_string(dis(gen));
  }

  std::cout << "Sequence: " << sequence << std::endl;
  return 0;
}

File: lab2/generator.java
import java.util.Random;
import java.io.FileWriter;
import java.io.IOException;

public class RandomGenerator {
    public static void main(String[] args) {
        // Генератор на основе вихря Мерсенна (аналог mt19937)
        Random gen = new Random();
        
        StringBuilder sequence = new StringBuilder();
        
        System.out.println("Сгенерированная последовательность (Java):");
        for (int i = 0; i < 128; i++) {
            int bit = gen.nextInt(2); // 0 или 1
            sequence.append(bit);
            System.out.print(bit);
        }
        System.out.println();
        
        // Сохранение в файл
        try {
            FileWriter writer = new FileWriter("sequence_java.txt");
            writer.write(sequence.toString());
            writer.close();
            System.out.println("\nПоследовательность сохранена в файл: sequence_java.txt");
        } catch (IOException e) {
            System.out.println("Ошибка при создании файла: " + e.getMessage());
        }
    }
}